cmd/bootstrap/utils/key_generation.go

```diff
@@ -173,7 +174,7 @@ func WriteMachineAccountFiles(chainID flow.ChainID, nodeInfos []bootstrap.NodeIn
 	//
 	// for the machine account key, we keep track of the address index to map
 	// the Flow address of the machine account to the key.
-	addressIndex := uint64(4)
+	addressIndex := uint64(systemcontracts.EVMStorageAccountIndex)
```

Comment:
Summarizing the machine account issue here:

Taking SN2 as an example, which has the following machine account file:
- Address: `0xb4bcb7e5ea8c4c64` [index 25]
- Machine account PubKey: `0x584319d80567e7b19414c1952a52b5e5a588d6170993ae84b4b655690233482c8fe9e354cf8e308b3771c04e80322b5b1dbe33fe4eeec67d8de6ea9d9dab6178`

On-chain we have:
- Account `0xb4bcb7e5ea8c4c64` [index 25] has the wrong key/hashing algo
- Account `0x50db0e38fb764cbb` [index 24] has pubkey `584319d80567e7b19414c1952a52b5e5a588d6170993ae84b4b655690233482c8fe9e354cf8e308b3771c04e80322b5b1dbe33fe4eeec67d8de6ea9d9dab6178`, matching the machine account file

This means that the key generation is using index 25 but the machine account was actually created in index 24. Meaning that our initial index is off by +1 (5 rather than 4). So I think reverting this change will resolve the problem (it also seems to indicate the `EVMStorageAccountIndex` constant is wrong.

```suggestion
	addressIndex := uint64(4)
```

Reply:
> it also seems to indicate the EVMStorageAccountIndex constant is wrong

That is concerning!

I will revert those changes from this PR and I will open another PR to deal with that.